hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -765,6 +767,34 @@ public void setEventTimeout(Duration timeout) {
       this.eventTimeout = timeout.toMillis();
     }
 
+    /**
+     * Deadline which should be set on commands sent from ReplicationManager
+     * to the datanodes, as a percentage of the event.timeout. If the command
+     * has not been processed on the datanode by this time, it will be dropped
+     * by the datanode and Replication Manager will need to resend it.
+     */
+    @Config(key = "command.deadline.factor",
+        type = ConfigType.DOUBLE,
+        defaultValue = "0.9",
+        tags = {SCM, OZONE},
+        description = "Fraction of the hdds.scm.replication.event.timeout "
+            + "from the current time which should be set as a deadline for "
+            + "commands sent from ReplicationManager to datanodes. "
+            + "Commands which are not processed before this deadline will be "
+            + "dropped by the datanodes. Should be a value > 0 and <= 1.")
+    private double commandDeadlineFactor = 0.9;
+    public double getCommandDeadlineFactor() {
+      return commandDeadlineFactor;
+    }
+
+    public void setCommandDeadlineFactor(double val) {
+      if (!(val > 0) || (val > 1)) {
+        throw new IllegalArgumentException(val
+            + " must be greater than 0 and less than equal to 1");
+      }
```

Comment:
Validation is better added to a separate method, annotated with `@PostConstruct`.  This ensures validation is performed even if the instance is created and initialized via reflections (also allows multiple config items to be cross-validated).

Some examples:

https://github.com/apache/ozone/blob/d0e6824eec7c015fb5066fefae89408220de1583/hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/OzoneClientConfig.java#L194-L220

https://github.com/apache/ozone/blob/d0e6824eec7c015fb5066fefae89408220de1583/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java#L171-L179

Reply:
Ah, I had a feeling there should be something like this. I will add this in.